lib/utils/job-utils/racadm-tool.js

```diff
@@ -192,29 +220,136 @@ function racadmFactory(
      * @param {string} host
      * @param {string} user
      * @param {string} password
-     * @param {Object} cifsConfig -  includes samba password, username and bios.xml file path
+     * @param {Object} cifsConfig -  includes samba password, username and image file path
+     * Password and username are only required when fiel path is a cifs share folder.
      */
-    RacadmTool.prototype.updateIdracImage = function(host, user, password, cifsConfig) {
-        var command = '';
-        var cifsUser = cifsConfig.user,
-            cifsPassword = cifsConfig.password;
-        assert.string(cifsUser);
-        assert.string(cifsPassword);
+    RacadmTool.prototype.updateFirmware = function(host, user, password, cifsConfig) {
+        var command = '', cifsUser, cifsPassword, self = this;
+        cifsConfig = cifsConfig || {};
+        if (!cifsConfig.filePath){
+            throw new Error('Can not find file path required for iDRAC image update');
+        }
+        cifsUser = cifsConfig.user || '';
+        cifsPassword = cifsConfig.password || '';
 
         var fileInfo = parser.getPathFilename(cifsConfig.filePath);
-        if (!_.endsWith(fileInfo.name, '.d7')){
-            throw new Error('iDRAC image format is not supported');
+        if (!(_.endsWith(fileInfo.name, '.d7') || _.endsWith(fileInfo.name, '.exe') ||
+            _.endsWith(fileInfo.name, '.EXE'))){
```

Comment:
As file extension is usually not case sensitive, so 'D7' or even 'eXe' should be OK as well. You can use a regex expression, so you don't have to list all possible cases.